### PR TITLE
Add Angular, Vue and React code snippets to dxToolbar

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 1]
```diff
@@ -76,6 +76,87 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
...
+                locate-in-menu="always">
+            </DxItem>
+            <DxItem
+                text="Change"
+                locate-in-menu="always">
+            </DxItem>
+            <DxItem
+                text="Products"
+                location="center">
+            </DxItem>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]: You can use self-closing tags in all these cases.

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 2]
```diff
@@ -76,6 +76,87 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
...
 
+##### Vue
+
+    <!--tab: App.vue-->
+    <template>
+        <DxToolbar>
+            <DxItem
+                widget="dxButton"
+                location="before"
+                :options="{
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]: Do not specify objects straight in the markup.

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 3]
```diff
@@ -76,6 +76,87 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
...
+                        locateInMenu="always">
+                    </Item>
+                    <Item
+                        text="Change"
+                        locateInMenu="always">
+                    </Item>
+                    <Item
+                        text="Products"
+                        location="center">
+                    </Item>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]: Same notes here

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md [hunk 4]
```diff
@@ -42,6 +45,53 @@ A **Toolbar** item may be plain text or a widget. Text items should have the [te
...
     })
 
+##### Vue
+
+    <!--tab: App.vue-->
+    <template>
+        <DxToolbar>
+            <DxItem text="Delete" location="before"></DxItem>
+            <DxItem text="Products" location="center"></DxItem>
+            <DxItem text="Add" location="after"></DxItem>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/05 Specify Item Type.md]: Self-closing tags.

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md [hunk 5]
```diff
@@ -123,6 +174,104 @@ Items that contain a widget should have the [widget](/api-reference/_hidden/dxTo
...
+                location="after">
+            </DxItem>
+        </DxToolbar>
+    </template>
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxToolbar, { DxItem } from 'devextreme-vue/toolbar';
+    import DxSelectBox from 'devextreme-vue/select-box';
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/05 Specify Item Type.md]:
```suggestion
    import 'devextreme-vue/select-box';
```

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md [hunk 6]
```diff
@@ -123,6 +174,104 @@ Items that contain a widget should have the [widget](/api-reference/_hidden/dxTo
...
+
+    export default {
+        components: {
+            DxToolbar,
+            DxItem,
+            DxSelectBox
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/05 Specify Item Type.md]:
```suggestion
            DxItem
```

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md (hunk 5 above)
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/05 Specify Item Type.md]: Don't you need to import the Button too?

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md (hunk 5 above)
Comment (EugeniyKiyashko) [reply concepts/05 Widgets/Toolbar/05 Specify Item Type.md]: It is works without import the Button

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 7]
```diff
@@ -65,7 +62,10 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
     import { DxToolbarModule, DxButtonModule } from "devextreme-angular";
     // ...
     export class AppComponent {
-        // ...
+        buttonOptions = {
+            type: 'back',
+            text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]:
```suggestion
            text: 'Back'
```

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 8]
```diff
@@ -76,6 +76,94 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
...
+    export default {
+        components: {
+            DxToolbar,
+            DxItem
+        },
+        data() {
+            return {
+                buttonOptions: {
+                    type: 'back',
+                    text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]:
```suggestion
                    text: 'Back'
```

File: concepts/05 Widgets/Toolbar/00 Overview.md [hunk 9]
```diff
@@ -76,6 +76,94 @@ The following code adds a simple **Toolbar** to your page. Three items are plain
...
+    <!--tab: App.js-->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Toolbar, Item } from 'devextreme-react/toolbar';
+
+    const buttonOptions = {
+        type: 'back',
+        text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/00 Overview.md]:
```suggestion
        text: 'Back'
```

File: concepts/05 Widgets/Toolbar/05 Specify Item Type.md [hunk 10]
```diff
@@ -123,6 +174,103 @@ Items that contain a widget should have the [widget](/api-reference/_hidden/dxTo
...
+    import { Toolbar, Item } from 'devextreme-react/toolbar';
+    import { SelectBox } from 'devextreme-react/select-box';
+
+    const buttonOptions = {
+        type: 'back',
+        text: 'Back',
+        onClick: function() {
+            // ...
+        } 
+    };
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/05 Specify Item Type.md]:
```suggestion
    }
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 11]
```diff
@@ -57,10 +58,13 @@ For a minor customization of **Toolbar** items, you can define [specific fields]
...
 
     <!--TypeScript-->
-    import { DxToolbarModule, DxButtonModule } from "devextreme-angular";
+    import { DxToolbarModule, DxButtonModule } from 'devextreme-angular';
     // ...
     export class AppComponent {
-        // ...
+        buttonOptions = {
+            type: 'back',
+            text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
            text: 'Back'
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 12]
```diff
@@ -71,31 +75,123 @@ For a minor customization of **Toolbar** items, you can define [specific fields]
...
+    export default {
+        components: {
+            DxToolbar,
+            DxItem
+        },
+        data() {
+            return {
+                buttonOptions: {
+                    type: 'back',
+                    text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
                    text: 'Back'
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 13]
```diff
@@ -71,31 +75,123 @@ For a minor customization of **Toolbar** items, you can define [specific fields]
...
+    <!--tab: App.js-->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Toolbar, Item } from 'devextreme-react/toolbar';
+
+    const buttonOptions = {
+        type: 'back',
+        text: 'Back',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
        text: 'Back'
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 14]
```diff
@@ -71,31 +75,123 @@ For a minor customization of **Toolbar** items, you can define [specific fields]
...
+            );
+        }
+    }
+
+    export default App;
+
 ---
 
-If you need a more flexible solution, define a custom template. For Angular, AngularJS and Knockout apps, DevExtreme provides the [dxTemplate](/api-reference/10%20UI%20Widgets/Markup%20Components/dxTemplate '/Documentation/ApiReference/UI_Widgets/Markup_Components/dxTemplate/') markup component. The following code gives a simple example of how you can use **dxTemplate** to customize items on the toolbar and commands on the overflow menu.
+If you need a more flexible solution, define an [itemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#itemTemplate') and [menuItemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#menuItemTemplate') callback functions to customize items on the toolbar and commands on the overflow menu. In Angular and Vue, you can declare it in the markup. In React, you can use a rendering function (shown in the code below) or component
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
If you need a more flexible solution, define an [itemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#itemTemplate') and [menuItemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#menuItemTemplate') to customize toolbar items and commands in the overflow menu, respectively. In Angular and Vue, you can declare the templates in the markup. In React, you can use rendering functions (shown in the code below) or components.
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 15]
```diff
@@ -117,106 +213,132 @@ If you need a more flexible solution, define a custom template. For Angular, Ang
...
+                }, {
+                    text: 'Change',
+                    locateInMenu: 'always'
+                }, {
+                    text: 'Remove',
+                    locateInMenu: 'always'
+                }, {
+                    text: 'Products',
+                    location: 'center'
+                }];
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
                }]
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 16]
```diff
@@ -117,106 +213,132 @@ If you need a more flexible solution, define a custom template. For Angular, Ang
...
+If you use jQuery, use <a href="http://api.jquery.com/category/manipulation/" target="_blank">DOM manipulation methods</a> to combine the HTML markup. To apply this markup to items on the toolbar and commands on the overflow menu, use the [itemTemplate](/api-reference/10%20UI%20Widgets/CollectionWidget/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#itemTemplate') and [menuItemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#menuItemTemplate') callback functions, respectively.
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
If you use jQuery, use <a href="http://api.jquery.com/category/manipulation/" target="_blank">DOM manipulation methods</a> to combine the HTML markup. To apply this markup to toolbar items and commands in the overflow menu, use the [itemTemplate](/api-reference/10%20UI%20Widgets/CollectionWidget/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#itemTemplate') and [menuItemTemplate](/api-reference/10%20UI%20Widgets/dxToolbar/1%20Configuration/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/#menuItemTemplate') callback functions, respectively.
```

File: concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md [hunk 17]
```diff
@@ -117,106 +213,132 @@ If you need a more flexible solution, define a custom template. For Angular, Ang
...
             }
         });
     });
 
+    <!--HTML-->
+    <div id="tabPanelContainer"></div>
+
+---
+
 You can also customize an individual toolbar item or menu command. For this purpose, declare a template for this item or command as a script and pass its `id` to the [template](/api-reference/_hidden/CollectionWidgetItem/template.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#template') or [menu item template](/api-reference/_hidden/dxToolbarItem/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#menuItemTemplate') option, respectively.
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Toolbar/15 Customize Item Appearance.md]:
```suggestion
You can also customize an individual toolbar item or menu command. For this purpose, declare a template for this item or command as a script and pass its `id` to the [template](/api-reference/_hidden/CollectionWidgetItem/template.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#template') or [menuItemTemplate](/api-reference/_hidden/dxToolbarItem/menuItemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#menuItemTemplate') option, respectively.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**